src/bech32.cpp

```diff
@@ -340,8 +344,8 @@ Encoding VerifyChecksum(const std::string& hrp, const data& values)
 /** Create a checksum. */
 data CreateChecksum(Encoding encoding, const std::string& hrp, const data& values)
 {
-    data enc = Cat(ExpandHRP(hrp), values);
-    enc.resize(enc.size() + 6); // Append 6 zeroes
+    auto enc = PreparePolynomialCoefficients(hrp, values, CHECKSUM_SIZE);
+    enc.insert(enc.end(), CHECKSUM_SIZE, 0x00);
     uint32_t mod = PolyMod(enc) ^ EncodingConstant(encoding); // Determine what to XOR into those 6 zeroes.
     data ret(6);
```

Comment:
instead of preallocation we could do a reserve here:
```C++
    data ret;
    ret.reserve(CHECKSUM_SIZE);
```

Reply:
Would prefer to keep this PR focused on the `limit` and `CHECKSUM_SIZE` change, but perhaps a good change for #29607